### PR TITLE
tslua: Exposes set/get method for server request objects

### PR DESCRIPTION
**Pull request by randall**

### CODE REVIEW

**Comment by randall** [comment]
[approve ci debian]

**Comment by zwoop** [comment]
Is there no documentation we should add to for these Lua methods?

**Comment by randall** [comment]
@zwoop Good catch! Added!

File: doc/admin-guide/plugins/lua.en.rst [hunk 1]
```diff
@@ -1903,6 +1903,26 @@ ts.server_request.set_url_scheme
 
 :ref:`TOP <admin-plugins-ts-lua>`
 
+ts.server_request.get_method
+----------------------------
+**syntax:** *ts.server_request.get_method()*
+
+**context:** do_remap/do_os_response or do_global_* or later
```
Comment (shukitchan) [inline doc/admin-guide/plugins/lua.en.rst]: this should only be allowed in TS_LUA_HOOK_SEND_REQUEST_HDR hook point

File: doc/admin-guide/plugins/lua.en.rst [hunk 2]
```diff
@@ -1903,6 +1903,26 @@ ts.server_request.set_url_scheme
...
+
+**description:** This function can be used to retrieve the current server request's method name. String like "GET" or "POST" is returned.
+
+:ref:`TOP <admin-plugins-ts-lua>`
+
+ts.server_request.set_method
+----------------------------
+**syntax:** *ts.server_request.set_method()*
+
+**context:** do_remap or do_global_*
```
Comment (shukitchan) [inline doc/admin-guide/plugins/lua.en.rst]: same comment as above

File: doc/admin-guide/plugins/lua.en.rst (hunk 1 above)
Comment (shukitchan) [reply doc/admin-guide/plugins/lua.en.rst]: pls check this out as example - https://docs.trafficserver.apache.org/en/latest/admin-guide/plugins/lua.en.html#ts-server-request-get-uri

**Comment by zwoop** [comment]
Cherry-picked to v9.0.x branch.

**Comment by zwoop** [comment]
Cherry-picked to 8.1.x

**Pull request merged**